src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1180,7 +1180,7 @@ public override BoundNode VisitCall(BoundCall node)
             VisitReceiverBeforeCall(node.ReceiverOpt, node.Method);
             VisitArgumentsBeforeCall(node.Arguments, node.ArgumentRefKindsOpt);
 
-            if (!callsAreOmitted && node.Method?.OriginalDefinition is LocalFunctionSymbol localFunc)
+            if (node.Method?.OriginalDefinition is LocalFunctionSymbol localFunc)
```

Comment:
Since `node` is actually a `BoundCall`, then we already know the local function is used. We want to visit it regardless it's **conditionally** omitted or not.

Reply:
I worry that this change may be unsound (allowing accessing an unassigned variable). Please add a test for:
```
public class C 
{
    public void M() 
    {
        int i;
        local();
        i.ToString();
        
        [Conditional("DEBUG")]
        void local()
        {
           i = 1;   
        }
    }
}
```

Reply:
It will be good to add this test. I believe it will produce the desired error on `i.ToString()` because we reset the definite assignment state after visiting the call when `callsAreOmitted` is true.

Reply:
Oh wait. I forgot that we disallow instance local functions from having `ConditionalAttribute` at all. We have [specified](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/local-function-attributes.md#attributes) that a local function must be `static` in order to be conditional.

Reply:
> we disallow non-static local functions from having ConditionalAttribute at all.

That saved some extra debugging :smile:

Reply:
@jcouv Please take another look at your convenience. I do not think there is a supported scenario involving capturing variables in a conditional local function.

Reply:
Sounds good. That's great.
Can we update the test (`LocalFunction_ConditionalAttributeDisallowed`). We should be able to remove the `#pragma warning`.